middleware/ratelimiter/store.go

```diff
@@ -0,0 +1,86 @@
+package ratelimiter
+
+import (
+	"math"
+	"sync"
+	"time"
+)
+
+type limiter struct {
+	requestQuota  int
+	counter       int
+	quotaDuration time.Duration
+	resetsAt      time.Time
+	mx            *sync.RWMutex
```

Comment:
```suggestion
	mx            sync.RWMutex
```
Don't use a pointer for mutex. After this change, you can also remove the initialization from `newLimiter()`